compile.ts

```diff
@@ -90,6 +91,28 @@ class Compiler {
       this.#rawEmit(".pname", reg, name);
       this.variable(param.name as ts.Identifier, reg);
     });
+    if (f.type) {
```

Comment:
Should this depend on a type declaration, or should we do this lazily when we see return with values?

Reply:
I am not sure to understand the question. Mostly I consider any input/output have to be Value (in the end they are registers), so I do not do any kind of type checking.